src/main/java/duke/Command.java

```diff
@@ -0,0 +1,252 @@
+                storage.rewriteFile(taskList.getTasks());
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "find":
+            if (args.charAt(0) == '"' && args.charAt(args.length() - 1) == '"') {
+                args = args.substring(1, args.length() - 1);
+            }
+            ArrayList<Task> matchingTasks = taskList.getMatchingTasks(args);
+            if (matchingTasks.isEmpty()) {
+                ui.print("No results found for keyword '" + args + "'");
+            } else {
+                ui.printTasks(matchingTasks, "Here are the matching tasks in your list:");
+            }
+            break;
+
+        case "list":
+            ui.printTasks(duke.getTaskList().getTasks(), "Here are the tasks in your list:");
+            break;
+
+        case "deadline":
+            String[] temp = args.split(" /by ", 2);
+            t = new Deadline(temp[0], LocalDate.parse(temp[1],
+                    DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT)));
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "event":
+            temp = args.split(" /at ", 2);
+            t = new Event(temp[0], LocalDate.parse(temp[1],
+                    DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT)));
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "todo":
+            t = new ToDo(args);
+            taskList.addTask(t);
+            ui.printTaskAdded(t, taskList.getCount());
+            try {
+                storage.appendTaskToFile(t);
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+
+        case "mark":
+
+        case "unmark":
+            index = Integer.parseInt(args) - 1;
+            t = taskList.getTasks().get(index);
+            boolean b = name.equals("mark");
+            taskList.markTask(t, b);
+            ui.printTaskMarked(t, b);
+            try {
+                storage.rewriteFile(taskList.getTasks());
+            } catch (IOException e) {
+                ui.printError("Unable to write to file.");
+            }
+            break;
+        }
+    }
+
+    private boolean isValidDate(String date) {
+        try {
+            LocalDate d = LocalDate.parse(date, DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT));
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
```

Comment:
You can consider adding header comments to all your methods to help developers better understand the code